premium/ecommerce/gdpr-send-to-mailchimp.php

```diff
@@ -5,38 +5,99 @@
+ * Check if a list member has status pending or subscribed based on there email
  *
  * @see https://mailchimp.com/developer/reference/lists/list-members/
  *
  * @param string $email
  *
  * @return bool
  */
-function sync_only_subscribed_or_pending_emails($email) {
+function mailchimp_list_member_has_status_pending_or_subscribed($email) {
+
 	if(!is_string($email)) {
 		throw new InvalidArgumentException(
-			sprintf("sync_only_subscribed_or_pending_emails() expects parameter 1 to be array or %s, %s given", Traversable::class, gettype($emails))
+			sprintf("mailchimp_list_member_has_status_pending_or_subscribed() expects parameter 1 to be string, %s given", gettype($email))
 		);
 	}
 
-	if(!mc4wp()->has('ecommerce.options')) {
-		return false;
-	}
+    if(!mc4wp()->has('ecommerce.options')) {
+        return false;
+    }
 
-	$options = mc4wp('ecommerce.options');
+    $options = mc4wp('ecommerce.options');
 
-	if (!isset($options['store']['list_id'])) {
-		return false;
-	}
+    if (!isset($options['store']['list_id'])) {
+        return false;
+    }
+    //get the list ecommerce. Make it static becuase
+    $list = $options['store']['list_id'];
 
-	$list =  $options['store']['list_id'];
 
 	try {
+	    //Send the request to get the status
 		$list_member = mc4wp_get_api_v3()->get_list_member($list, $email);
 	} catch (Exception $e) {
+	    //In case of an exception return false
 		return false;
 	}
 
-	return in_array($list_member->status, array('subscribed', 'pending'));
+    //Check the status
+	return in_array($list_member->status, array('subscribed', 'pending'), true);
 }
+
+/**
+ * Only is called whe
+ *
+ * @param
+ */
+add_filter('mc4wp_ecommerce_send_cart_to_mailchimp', function ($value, $customer) {
+    $email = null;
+
+    switch (get_class($customer)) {
+        case stdClass::class:
+            $email = $customer->billing_email;
+            break;
+        case WP_User::class:
+            $email = $customer->user_email;
+            break;
+        case WC_Customer::class:
+            $email = $customer->get_email();
+            break;
+    }
+
+    if (! $email) {
+        return false;
+    }
+
+	return mailchimp_list_member_has_status_pending_or_subscribed($email);
+}, 10, 2);
+
+/**
+ * This filter will be called when a client updates his profile.
+ *
+ * @param bool $value
+ *
+ * @see ecommerce3/includes/class-ecommerce.php#186
+ */
+add_filter('mc4wp_ecommerce_send_customer_to_mailchimp', function ($value, $customer) {
+    //WP_user if user is changed
+    if (! $customer instanceof WP_User ||
+        ! $customer->has_prop('user_email')
+    ) {
+        return false;
+    }
+
+	return mailchimp_list_member_has_status_pending_or_subscribed(
+	    $customer->get('user_email')
+    );
+}, 10, 2);
+
+add_filter('mc4wp_ecommerce_send_order_to_mailchimp', function ($value, WC_Order $order) {
+    //If the order
+    if ($order->has_shipping_address()) {
```

Comment:
Hmmm, wouldn't this rule out any customers that do not provide shipping details (eg for digital products this often isn't needed)? I think we should only check for a valid `billing_email` property here.